HIP/hip-0000-cross-ledger-protocol-clpr.md

```diff
@@ -0,0 +1,1152 @@
+application logic and the mechanics of sending messages via state proofs of
+connector state. Combining the use of state proofs to establish message
+order with the semantics of guaranteed, exactly once, in-order message
+handling, Clipper is able to preserve the asynchronous Byzantine
+Fault Tolerant (aBFT) security profile of an already aBFT secure ledger.
+
+![Separable Connector and Application](../assets/hip-0000-clpr/clpr-simple-high-level-diagram.drawio.svg)
+
+Two Clipper enabled ledgers can communicate directy with each other without
+the use of intermediate relays, bridges, or oracles. If both ledgers are
+aBFT, then the combined interaction is aBFT. If one ledger was
+only BFT while the other was aBFT, the combined interaction is BFT. Clipper
+preserves the weaker security profile of the interacting ledgers.
+
+Clipper is extensible in two ways. The `Clipper Application Protocol`
+specifies the space of possible application use cases and can can be
+extended to with new sets of messages to enable new use cases. The
+`Clipper Connector Protocol` can be extended to support new ledger types and
+state proof formats.
+
+In this HIP we specify the core behavior of a generic `Clipper Endpoint`,
+the core mechanics of a `Clipper Application`, and Hiero's and
+implementation of both.
+
+## Glossary of Terms
+
+* `Clipper`
+    * A framework and collection of protocols for enabling interledger
+      applications to communicate through aBFT secure message passing via the
+      connector abstraction.
+* `Clipper Application`
+    * A distributed application installed on two or more ledgers that passes
+      messages through a connector to communicate. A clipper application's
+      functionality is organized into one or more use cases.
+* `Application Message`
+    * A messages sent from one application installed on one ledger to a
+      remote application installed on a different ledger through a connector
+      between the two ledgers.
+* `Clipper Application Protocol`
+    * The Clipper Application Protocol is the specification of application
+      messages, callback APIs, and semantics for asynchronously handling the
+      messages, all organized into discrete use cases.
+* `Connector`
+    * A connector manages the outgoing application message queue, permission to
+      submit messages to the queue, and payment mechanics for handling sent and
+      received application messages.
+* `Connector State Proof`
+    * A state proof that establishes the state of a connector, indicating the
+      messages received and sent, or provides proof of a portion of the outgoing
+      message queue contents of a connector.
+* `Clipper Endpoint`
+    * A clipper endpoint is a network server endpoint (ip address and port)
+      for a machine that can share state proofs for one or more connectors.
+      There may be multiple clipper endpoints for a ledger, but there is a
+      single ledger associated with each clipper endpoint.
```

Comment:
There is one or more connector per endpoint, but multiple endpoints per ledger -- I assume this is because each node in the ledger can run its own endpoint, but any given node would (usually?) have a single endpoint?

Reply:
How do we deal with the fact that ledger endpoints could change daily, or even more frequently?